src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -282,7 +282,7 @@ public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE
         // calling $this->normalizeId($id) unless necessary.
         for ($i = 2;;) {
             if (isset($this->privates[$id])) {
-                @trigger_error(sprintf('Requesting the "%s" private service is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0.', $id), E_USER_DEPRECATED);
+                @trigger_error(sprintf('The "%s" service is private, getting it from the container is deprecated since Symfony 3.2 and will fail in 4.0. You should either make the service public, or preferably eg implement ServiceSubscriberInterface or use any other dependency injection method instead of using the container at all.', $id), E_USER_DEPRECATED);
```

Comment:
This is the most critical one, as it will be triggered quite often in 3.4 where services are now private by default.

Reply:
Minor reword proposal:

Before:

```
The "%s" service is private, getting it from the container is deprecated since
Symfony 3.2 and will fail in 4.0. You should either make the service public, or
preferably eg implement ServiceSubscriberInterface or use any other dependency
injection method instead of using the container at all.
```

After:

```
The "%s" service is private, getting it from the container is deprecated since
Symfony 3.2 and will fail in 4.0. You should either make the service public or
stop getting services directly from the container (instead, implement
ServiceSubscriberInterface or use other dependency injection methods).
```

Reply:
I would not mention `ServiceSubscriberInterface` here as this is just one example amongst many other ones.

Reply:
updated, better?